docs/SQL_SERVER_WC.md

```diff
@@ -0,0 +1,72 @@
+### SQL Server limitations
+
+Almost all geospatial data can be converted to SQL Server format without losing any fidelity, but it does have the following limitations.
+
+#### Approximated types
+
+There is one type that Sno supports that has no SQL Server equivalent - the `interval`. This type is "approximated" as a string (or more precisely, as an `NVARCHAR`) in the SQL Server working copy while keeping its original type in the Sno dataset. Sno creates a column of type `NVARCHAR` in the appropriate place in the database table, and fills it with with the intervals formatted as [ISO8601 durations](https://en.wikipedia.org/wiki/ISO_8601#Durations). When the working copy is committed, Sno converts the contents of those columns back to `interval`, instead of their actual type, `NVARCHAR`. Since the change from `interval` to `NVARCHAR` is just a limitation of the working copy, this apparent change in type will not show up as a change in the commit log, nor will it show up as an uncommitted change if you run `sno status` to see what local changes you have made but not yet committed.
+
+#### Geometry types
+
+Sno lets you define a column as containing only a particular type of geometry, eg only `POINT` or only `MULTIPOLYGON` types. By contrast, SQL Server lets you put any type of geometry into a geometry column.
+
+This mismatch has the following consequence: If Sno is managing a geometry column with a particular geometry type such as `POINT`, and you check it out in a SQL Server working copy, you will be able to insert other types of geometry into it, but Sno will prevent you from committing it. You still need to follow the constraint put in place when the dataset was created, and only insert new geometries of the appropriate type.
+
+If you need decide that a certain dataset should contain more types of geometries than its constraint currently allows, it is possible to change a columns geometry type to be broader and allow more types. This cannot be done by editing the SQL Server working copy, since as noted it doesn't store this type information - instead it must be done using either a different type of working copy, or the Sno command line tool. To use the command line, take the following steps:
+
+1. View all the metadata for your dataset:
+   `sno meta get DATASET`
+2. Copy the JSON from under the heading `schema.json` and save it to a file of the same name in your current working directory.
+3. Modify the JSON so that the `geometryType` property is broader. For example:
+   - Old line: `  "geometryType": "POINT",`
+   - New line: `  "geometryType": "GEOMETRY",`
+4. Commit this change to the schema:
+   `sno meta set DATASET schema.json=@schema.json`
+
+#### CRS definitions
+
+Sno lets you define arbitrary CRS definitions and attach them to your dataset. By contrast, SQL Server comes pre-installed with hundreds of the standard EPSG & ESRI coordinate reference system definitions. However, these cannot be modified, and custom CRS cannot be added.
```

Comment:
AFAICT SQL Server is naive to CRS, with no CRS definitions or ability to transform coordinates between them. SRIDs are just reference numbers.

Reply:
This is not quite true but I'm not 100% on the details. Here are two counterpoints I do know:
- There is a table sys.spatial_reference_systems, which stores the supported CRS definitions. Now, this doesn't necessarily mean that SQL server actually cares about those definitions, but...
- A bunch of Geography type instance methods use them. For instance, STArea:
https://docs.microsoft.com/sql/t-sql/spatial-geography/starea-geography-data-type
But also STDistance, and things like intersection and union - which also require that the types they are unioning or intersecting have matching SRIDs.

But looking into it more, seems like this mostly doesn't matter to us yet as we are using Geometry types? It matters that Geometries have matching SRIDs, but it doesn't seem to matter what they are - eg the area is always given in the same units as the geometries themlselves, which exist on an abstract flat plane. So, basically you are right about SRIDs as long as we are talking about Geometries and not Geographies.

I'll rewrite it a bit.

Reply:
Let’s get it right for the code we have and we can expand further for Geography soon.